cmd/login/keypair.go

```diff
@@ -0,0 +1,183 @@
+package login
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"time"
+
+	corev1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/afero"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"github.com/giantswarm/kubectl-gs/pkg/data/domain/keypair"
+)
+
+const (
+	credentialRetryTimeout    = 250 * time.Millisecond
+	credentialMaxRetryTimeout = 2 * time.Second
+
+	credentialKeyCertCRT = "crt"
+	credentialKeyCertKey = "key"
+	credentialKeyCertCA  = "ca"
+)
+
+func generateKeypairUID() string {
+	hash := sha256.New()
+	_, _ = hash.Write([]byte(time.Now().String()))
+
+	uid := fmt.Sprintf("%x", hash.Sum(nil))
+
+	return uid[:16]
+}
+
+func generateKeypair(cluster, organization, ttl string, groups []string, clusterBasePath string) (*keypair.Keypair, error) {
+	keypairUID := generateKeypairUID()
+	keypairName := fmt.Sprintf("%s-%s", cluster, keypairUID)
+	commonName := fmt.Sprintf("%s.%s.k8s.%s", keypairUID, cluster, clusterBasePath)
+
+	certConfig := &corev1alpha1.CertConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      keypairName,
+			Namespace: metav1.NamespaceDefault,
+			Labels: map[string]string{
+				"cert-operator.giantswarm.io/version": "0.1.0",
```

Comment:
Can this always stay `0.1.0`? I think we have different versions of `cert-operator` or am I mistaken?